### PR TITLE
Flow cleanup

### PR DESCRIPTION
**Pull request by MilanBarande**
💥 Flow fixes for BrightMirror, Announcement, textAndMedia && topPostFormContainer 💥

### CODE REVIEW

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 1]
```diff
@@ -42,10 +43,21 @@ export type AnnouncementContent = {
   __typename?: string
 };
 
+type MessageColumns = Array<IdeaMessageColumnFragment>;
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: We already have IdeaMessageColumns in object_types.js

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 2]
```diff
@@ -72,12 +84,16 @@ export const getSentimentsCount = (posts: Posts) => {
...
-    count: sentimentCounts[key].count,
-    Tooltip: createTooltip(sentimentCounts[key], sentimentCounts[key].count)
-  }));
+export const createDoughnutElements = (sentimentCounts: SentimentsCounts): Array<DoughnutElements> =>
+  Object.keys(sentimentCounts).map(
+    key =>
+      sentimentCounts[key] && {
+        color: sentimentCounts[key].color,
+        count: sentimentCounts[key].count,
+        // $FlowFixMe flow believes sentimentCounts[key] can be undefined but it's not annotated as possibly undefined
```
Comment (cedricmessiant) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: yeah, I guess that flow don't see that you are currently iterate over keys and thus `sentimentCounts[key]` can't be undefined

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 3]
```diff
@@ -18,8 +18,9 @@ export const createTooltip = (sentiment: SentimentDefinition, count: number) =>
 
 type SentimentsCounts = {
   [string]: {
-    ...SentimentDefinition,
-    count: number
+    count: number,
+    color: string,
+    Tooltip: React.Node
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: This change is wrong. You need the full `...SentimentDefinition` here, cameType is used in `createTooltip` via `createDoughnutElements`

File: assembl/static2/js/app/components/debate/common/announcement.jsx (hunk 2 above)
Comment (vincentfretin) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: Do you still need this FlowFixMe with the `sentimentCounts[key]` check above and after reverting the SentimentsCounts changes?

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 4]
```diff
@@ -133,6 +149,7 @@ const Announcement = ({ idea, announcementContent, isMultiColumns }: Props) => {
             </div>
           ) : (
             <div className="announcement-numbers">
+              {/* $FlowFixMe numContributors is declared as a number in Idea (object_types) but flow believes it can be undefined */}
```
Comment (cedricmessiant) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: strange indeed

File: assembl/static2/js/app/components/common/textAndMedia.jsx [hunk 5]
```diff
@@ -104,16 +103,17 @@ class TextAndMedia extends React.Component<Props> {
             <Grid fluid>
               {somethingOnLeft && (
                 <Col sm={totalSize} md={somethingOnRight ? leftSize : totalSize}>
-                  {validDescriptionSide && <TextAndMedia.SideDescription content={descriptionSide} />}
+                  {descriptionSide && validDescriptionSide && <TextAndMedia.SideDescription content={descriptionSide} />}
```
[the remark this answers is not in the document]
Comment (MilanBarande) [reply assembl/static2/js/app/components/common/textAndMedia.jsx]: It doesn't work. This is the typical ugly check flow forces you to do sometimes 😕

File: assembl/static2/js/app/components/debate/common/announcement.jsx (hunk 3 above)
Comment (MilanBarande) [reply assembl/static2/js/app/components/debate/common/announcement.jsx]: even though SentimentDefinition declaration in sentimentDefinitions.jsx declares ```color: string``` down in createDoughtnutElements there is an error on ```color: sentimentCounts[key].color``` if I don't explicitely declare it in type SentimentCounts :( 
I think flow has trouble with mapping over Object.keys

File: assembl/static2/js/app/components/debate/common/announcement.jsx (hunk 2 above)
Comment (MilanBarande) [reply assembl/static2/js/app/components/debate/common/announcement.jsx]: Yes

Comment (MilanBarande) [reply assembl/static2/js/app/components/debate/common/announcement.jsx]: Yes, there are always issues with flow with this kind of iteration, I hope they'll fix it

**Pull request merged**